src/DN404.sol

```diff
@@ -462,6 +468,7 @@ abstract contract DN404 {
                         if (_useExistsLookup()) _set($.exists, id, true);
                         _set(toOwned, toIndex, uint32(id));
                         _setOwnerAliasAndOwnedIndex(oo, id, t.toAlias, uint32(toIndex++));
+                        _afterNFTMint(id);
```

Comment:
```solidity
/// @dev Hook that is called after any NFT token transfers, including minting and burning.
function _afterNFTTransfer(address from, address to, uint256 id) internal virtual {}
```

Would prefer a single function. Not sure if will trigger stack-too-deep.

Reply:
Yes. Changing to `_afterNFTTransfer` per suggestion initially leads to stack-too-deep.

```
Error: Compiler error (/solidity/libsolidity/codegen/LValue.cpp:55):Stack too deep. Try compiling with `--via-ir` (cli) or the equivalent `viaIR: true` (standard JSON) while enabling the optimizer. Otherwise, try removing local variables.
   --> src/DN404.sol:601:39:
    |
601 |                     _afterNFTTransfer(from, address(0), id);
```

I added a scope to save one local variable in `_burn` to fix the stack-too-deep issue. Should not change any behavior.

Reply:
Very good idea.